File: src/main/java/com/windchopper/fs/SftpDirectoryStream.java
package com.windchopper.fs;

import java.nio.file.DirectoryStream;
import java.nio.file.Path;
import java.util.Iterator;
import java.util.List;

public class SftpDirectoryStream implements DirectoryStream<Path> {

    private final List<Path> paths;

    SftpDirectoryStream(List<Path> paths) {
        this.paths = paths;
    }

    @Override public Iterator<Path> iterator() {
        return paths.iterator();
    }

    @Override public void close() {
        // nothing to do
    }

}

File: src/main/kotlin/com/github/windchopper/fs/sftp/SftpDirectoryStream.kt
package com.github.windchopper.fs.sftp

import java.nio.file.DirectoryStream
import java.nio.file.Path

class SftpDirectoryStream(filter: DirectoryStream.Filter<in Path?>, paths: List<Path>): DirectoryStream<Path> {

    private val filteredPaths: MutableList<Path> = paths
        .filter { path -> filter.accept(path) }
        .toMutableList()

    override fun iterator(): MutableIterator<Path> {
        return filteredPaths.iterator()
    }

    override fun close() {
        // nothing to do
    }

}